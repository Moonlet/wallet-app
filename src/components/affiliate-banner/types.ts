export const AffiliateBannerType = {
    LEDGER_NANO_X: {
        url: 'https://shop.ledger.com/pages/ledger-nano-x?r=69d534ac55d9',
        image: require('../../assets/images/png/affiliate-banners/ledger_nano_x.png')
    },
    UNSTOPPABLE_DOMAINS: {
        url: 'https://unstoppabledomains.com/r/moonlet',
        image: require('../../assets/images/png/affiliate-banners/unstoppabledomains.png')
    }
};

export interface IAffiliateBannerType {
    url: string;
    image: any;
}
