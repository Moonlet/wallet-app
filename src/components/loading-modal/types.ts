import { TransactionMessageType, TransactionMessageText } from '../../core/blockchain/types';

export interface ILoadingModalMessage {
    type: TransactionMessageType;
    text?: TransactionMessageText;
    component?: any; // React.ComponentType<any>;
}

export interface ILoadingModalState {
    isVisible: boolean;
    message: ILoadingModalMessage;
}
