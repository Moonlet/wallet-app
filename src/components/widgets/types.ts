import { Blockchain } from '../../core/blockchain/types';
import { handleCta } from '../../redux/ui/screens/data/handle-cta';
import { AccountType } from '../../redux/wallets/state';
import { IconValues } from '../icon/values';
import {
    clearScreenInputData,
    runScreenValidation,
    runScreenStateActions,
    setScreenInputData
} from '../../redux/ui/screens/input-data/actions';

export interface IScreenRequest {
    context: IScreenContext;
    user: IScreenUser;
}

export interface IScreenContext<ScreenParams = any> {
    screen: string;
    step?: string;
    tab?: string;
    flowId?: string;
    params?: ScreenParams;
    flowData?: any;
}

export interface IScreenCtaContextParams<F = any, S = any, E = any> {
    ctaId: string;
    flowInputData: F;
    screenInputData: S;
    extraParams?: E;
}

/** @deprecated use string instead */
export enum ContextScreen {
    DASHBOARD = 'dashboard',
    TOKEN = 'token',
    QUICK_STAKE_SELECT_VALIDATOR = 'quickStakeSelectValidator'
}

/** @deprecated use string instead */
export enum ContextTab {
    ACCOUNT = 'account'
}

export interface ISmartScreenActions {
    handleCta?: typeof handleCta;
    clearScreenInputData?: typeof clearScreenInputData;
    runScreenValidation?: typeof runScreenValidation;
    runScreenStateActions?: typeof runScreenStateActions;
    setScreenInputData?: typeof setScreenInputData;
}

export enum SmartScreenPubSubEvents {
    SCROLL_TO_END = 'scrollToEnd',
    COLLAPSE_ALL = 'collapseAll'
}

export interface IScreenUser {
    os: 'android' | 'ios' | 'web';
    deviceId: string;
    preferedCurrency: string;
    appVersion: string;
    theme: 'dark' | 'light';
    country?: string;
    lang: 'en';

    wallet: {
        pubKey: string;
        type: 'HD' | 'HW';
        hwOptions?: {
            vendor?: string;
            model?: string;
            connectionType?: string;
        };
    };

    blockchain: Blockchain;
    chainId: string;
    address: string;
    accountType: AccountType;
}

export interface IScreenResponse {
    widgets: IScreenWidget[];
    bottomFixedArea?: IScreenWidget;
    validation?: IScreenValidation;
    navigationOptions?: any;
}

export interface IScreenCtaResponse {
    cta: ICta;
}

export interface IScreenFieldValidation {
    fn: string;
    params?: any[];
    messages?: {
        [key: string]: {
            type: string;
            message: string;
        };
    };
}

export interface IScreenValidation {
    validators: {
        [field: string]: IScreenFieldValidation[];
    };
}

export interface ICtaAction<P = any> {
    type: 'callAction' | 'openUrl' | 'navigateTo' | 'onBack' | 'navigationOnPop';
    params: {
        action?: string;
        url?: string;
        count?: string;
        screen?: string;
        params?: P;
    };
}

export interface ICta {
    actions?: ICtaAction[];

    /** @deprecated use actions instead */
    type?: 'callAction' | 'openUrl' | 'navigateTo';

    /** @deprecated use actions instead */
    params?: {
        action?: string;
        url?: string;
        screen?: string;
        params?: any;
    };

    label?: string;

    buttonProps?: {
        primary?: boolean;
        secondary?: boolean;
        disabled?: boolean;
        disabledSecondary?: boolean;

        colors?: {
            /** @deprecated use textStyle instead */
            label?: string;

            bg?: string;
        };

        leftIcon?: IconValues;

        wrapperStyle?: any;
        buttonStyle?: any;
        textStyle?: any;
    };

    // TODO: check this
    screenDataValidation?: {
        context: IScreenContext; // not sure if needed
    };
}

export interface IScreenWidget {
    title?:
        | string
        | {
              value: string;
              containerStyle?: IDataStyle;
              textStyle?: IDataStyle;
              iconStyle?: IDataStyle;
          };
    expandable?: boolean;
    initialState?: 'collapsed' | 'expanded';
    style?: IDataStyle;
    modules: IScreenModule[];
    cta?: ICta;
    hideExpandableIcon?: boolean;
}

export interface IScreenModule {
    displayWhen?: 'collapsed' | 'expanded'; // if undefined, it will be displayed always
    hidden?: boolean;
    style?: IDataStyle;
    type:
        | ModuleTypes.ABSOLUTE_MODULES
        | ModuleTypes.AMOUNT_INPUT
        | ModuleTypes.AMOUNT_SELECTABLE_BOX
        | ModuleTypes.BALANCES_GRID_ICONS
        | ModuleTypes.CTA
        | ModuleTypes.GRADIENT_WRAPPER
        | ModuleTypes.ICON_ONE_LINE
        | ModuleTypes.ICON
        | ModuleTypes.ICON_TWO_LINES
        | ModuleTypes.IMAGE_BANNER
        | ModuleTypes.INPUT
        | ModuleTypes.TIMER_INTERVAL_PRICE_UPDATE
        | ModuleTypes.MD_TEXT
        | ModuleTypes.MODULE_COLUMNS_WRAPPER
        | ModuleTypes.MODULE_SELECTABLE_WRAPPER
        | ModuleTypes.MODULE_WRAPPER
        | ModuleTypes.ONE_LINE_TEXT_BANNER
        | ModuleTypes.PRICE_UPDATE
        | ModuleTypes.PROGRESS_BAR
        | ModuleTypes.SEPARATOR
        | ModuleTypes.SINGLE_BALANCE_ICON
        | ModuleTypes.STATIC_TEXT_COLUMNS_BOTTOM_HEADER
        | ModuleTypes.STATIC_TEXT_COLUMNS_TOP_HEADER
        | ModuleTypes.TEXT_LINE_ICON
        | ModuleTypes.THREE_LINES_CTA
        | ModuleTypes.THREE_LINES_ICON
        | ModuleTypes.TWO_LINES_TEXT_BANNER
        | ModuleTypes.VALIDATIONS;
    cta?: ICta;
    data:
        | I2LinesTextBannerData
        | I3LinesCtaData
        | IAmountInputData
        | IAmountSelectableBoxData
        | IBalanceGridData
        | IBalanceGridData[]
        | IGradientWrapperData
        | IIconData
        | IIconTwoLinesData
        | IIconOneLineData
        | IImageBannerData
        | IInputData
        | IMdTextData
        | IOneLineTextBannerData
        | IScreenModuleColumnsWrapperData
        | IScreenModuleSelectableWrapperData
        | IScreenModuleWrapperData
        | ISeparatorData
        | IStaticTextColumnData[]
        | IThreeLinesIconData;
    details?: any; // ex. amount, validatorId, ...
    info?: {
        position?: string; // 'top-right' is default
        style?: IDataStyle;
        data: IScreenModule;
    };

    state: {
        // uiStateSelector?: IStateSelector, // aka state modifier - not implemented now
        actions?: IStateSelector[];
        selectors: {
            [key: string]: IStateSelector;
        };
    };
}

export interface IStateSelector {
    fn: string;
    params?: any[];
}

export enum ModuleTypes {
    ABSOLUTE_MODULES = 'absolute-modules',
    AMOUNT_INPUT = 'amount-input',
    AMOUNT_SELECTABLE_BOX = 'amount-selectable-box',
    BALANCES_GRID_ICONS = 'balances-grid-icons',
    CTA = 'cta',
    GRADIENT_WRAPPER = 'gradient-wrapper',
    ICON_ONE_LINE = 'cta-one-line',
    ICON = 'icon',
    ICON_TWO_LINES = 'icon-two-lines',
    IMAGE_BANNER = 'image-banner',
    INPUT = 'input',
    TIMER_INTERVAL_PRICE_UPDATE = 'timer-interval-price-update',
    MD_TEXT = 'md-text',
    MODULE_COLUMNS_WRAPPER = 'module-columns-wrapper',
    MODULE_SELECTABLE_WRAPPER = 'module-selectable-wrapper',
    MODULE_WRAPPER = 'module-wrapper',
    ONE_LINE_TEXT_BANNER = 'one_line_text_banner',
    PRICE_UPDATE = 'price-update',
    PROGRESS_BAR = 'progress-bar',
    SEPARATOR = 'separator',
    SINGLE_BALANCE_ICON = 'single-balance-icon',
    STATIC_TEXT_COLUMNS_BOTTOM_HEADER = 'static-text-columns-bottom-header',
    STATIC_TEXT_COLUMNS_TOP_HEADER = 'static-text-columns-top-header',
    TEXT_LINE_ICON = 'text-line-icon',
    THREE_LINES_CTA = '3-lines-cta',
    THREE_LINES_ICON = 'three-lines-icon',
    TWO_LINES_TEXT_BANNER = '2-lines-text-banner',
    VALIDATIONS = 'validations'
}

/// Screen Module Data Types \\\

export interface IScreenModuleWrapperData {
    state: string; // DEFAULT | PENDING | HIDDEN
    stateModifierFn: string;
    data: {
        [state: string]: Partial<IScreenModule>;
    };
}

export interface IScreenModuleSelectableWrapperData {
    state: string; //
    stateModifierFn: string;
    style?: {
        [state: string]: IDataStyle;
    };
    submodules: IScreenModule[];
}

export interface IScreenModuleColumnsWrapperData {
    style?: IDataStyle;
    submodules: IScreenModule[];
}

export enum DataType {
    TEXT = 'TEXT',
    CURRENCY = 'CURRENCY'
}

export interface ITextData {
    value: string | number;
    params?: {
        [param: string]: IData;
    };
}

export interface ICurrencyData {
    value: string;
    symbol: string;
    decimals?: number;
    round?: {
        decimals: number;
        type?: 'UP' | 'DOWN';
    };
    beautify?: IBeautify;
}

export interface IBeautify {
    fromValue?: number;
    decimals?: number;
    symbol?: string;
    notation?: 'compact' | 'percent';
}

export interface IDataStyle {
    [key: string]:
        | string
        | number
        | {
              value: string | number;
              fn: 'normalize' | 'normalizeFontAndLineHeight' | 'gradient';
          };
}

export interface IData {
    type: DataType;
    style?: IDataStyle;
    data: ITextData | ICurrencyData;
}

export interface I3LinesCtaData {
    firstLine: IData[];
    secondLine: IData[];
    thirdLine: IData[];
}

// Used for `static-text-columns-top-header` and `static-text-columns-bottom-header`
export interface IStaticTextColumnData {
    header: IData[];
    body: IData[];
}

// Used for`balances-grid-icons` and `single-balance-icon`
export interface IBalanceGridData {
    balance: IData[];
    label?: IData[];
    icon: {
        value: IconValues | string;
        color: string;
    };
}

export interface IImageBannerData {
    image: IImageInfo;
}

export interface IImageInfo {
    url: string;
    width: number;
    height: number;
}

export interface IOneLineTextBannerData {
    line: IData[];
}

export interface I2LinesTextBannerData {
    firstLine: IData[];
    secondLine: IData[];
    icon?: {
        value: IconValues;
        color?: string;
    };
    backgroundColor?: string;
}

export interface ISeparatorData {
    color?: string;
}

export interface IIconTwoLinesData {
    firstLine: IData[];
    secondLine: IData[];
    icon?: {
        url: string;
        style?: IDataStyle;
    };
}

export interface IIconData {
    icon: IconValues;
}

export interface IPriceUpdateData {
    endpoint: {
        url: string;
        method: 'POST' | 'GET';
        data: any;
    };
    reduxKey: string;
    interval: number; // miliseconds
}

export interface ITimerUpdateData {
    numSeconds: number;
    reduxKey: string;
    endpoint: {
        url: string;
        method: 'POST' | 'GET';
        data: any;
    };
    cta: ICta;
    interval: number; // miliseconds
}

export interface IThreeLinesIconData {
    firstLine: IData[];
    secondLine: IData[];
    thirdLine: IData[];
    icon: {
        url: string;
        style?: IDataStyle;
    };
}

export interface IMdTextData {
    text: string;
    style?: any;
}

/**
 * value: +0.1 | half | all
 * percentage: 10%
 */
export interface IAmountInputAmountBox {
    type: 'value' | 'percentage';
    value: string | number;
    // label: string;
}

export interface IAmountInputData {
    input?: {
        style?: IDataStyle;
        textStyle?: IDataStyle;
    };
    labels?: IData[];
    amounts?: IAmountInputAmountBox[];
    placeholder?: {
        value: string;
        color?: string;
    };
    editable?: boolean;
    focus?: boolean;
    showValidations?: boolean;
    onChangeTextAction?: IStateSelector;
}

export interface IAmountSelectableBoxData {
    amounts: IAmountInputAmountBox[];
}

export interface IIconOneLineData {
    icon: {
        value: IconValues;
        style?: IDataStyle;
    };
    line: IData[];
}

export interface IInputData {
    style?: {
        input?: IDataStyle;
        inputContainer?: IDataStyle;
    };
    options?: {
        keyboardType?: string;
        placeholder?: {
            value: string;
            color?: string;
        };
        multiline?: boolean;
    };
}

export interface IGradientWrapperData {
    gradient: string[];
    submodules: IScreenModule[];
}

export interface IProgressBarData {
    percentage: number;
    backgroundStyle?: IDataStyle;
    barStyle?: IDataStyle;
}

export interface ITextLineIconData {
    icon: {
        value: IconValues;
        style?: IDataStyle;
    };
    line: IData[];
    lineStyle?: IDataStyle;
}

export interface IValidationData {
    fieldName?: string;
}

export interface IAbsoluteModulesData {
    module1: {
        module: IScreenModule;
        style?: IDataStyle;
    };
    module2: {
        module: IScreenModule;
        style?: IDataStyle;
    };
}
