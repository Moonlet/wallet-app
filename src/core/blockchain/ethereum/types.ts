export enum MethodSignature {
    INCREASE_ALLOWANCE = 'increaseAllowance(address,uint256)',
    DELEGATE = 'delegate(address,uint256)',
    UNDELEGATE = 'undelegate(address,uint256)',
    WITHDRAW_DELEGATED = 'withdrawDelegated(address,address)',
    TRANSFER = 'transfer(address,uint256)'
}
