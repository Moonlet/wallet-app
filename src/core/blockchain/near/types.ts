import BigNumber from 'bignumber.js';
import { PosBasicActionType } from '../types/token';

export enum NearTransactionActionType {
    TRANSFER = 'TRANSFER',
    FUNCTION_CALL = 'FUNCTION_CALL'
}

export enum NearFunctionCallMethods {
    DEPOSIT = 'deposit',
    STAKE = 'stake',
    UNSTAKE = 'unstake',
    WITHDRAW = 'withdraw'
}

export interface INearTransactionAction {
    type: NearTransactionActionType;
    params?: any[];
}

export interface INearTransactionAdditionalInfoType {
    currentBlockHash: string;
    actions: INearTransactionAction[];
    posAction?: PosBasicActionType;
    validatorName?: string;
}

export enum NearAccountType {
    DEFAULT = 'DEFAULT',
    CONTRACT = 'CONTRACT'
}

export interface INearAccount {
    address: string;
    name: string;
    amount: BigNumber;
    exists: boolean;
    valid: boolean;
    type?: NearAccountType;
}
