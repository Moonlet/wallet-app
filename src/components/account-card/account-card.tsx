import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Text } from '../../library';
import { IAccountState } from '../../redux/wallets/state';
import { Blockchain } from '../../core/blockchain/types';
import { BLOCKCHAIN_INFO } from '../../core/constants/blockchain';
import { Icon } from '../icon';
import { Convert } from '../convert/convert';

import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import { NavigationScreenProp, NavigationState, NavigationParams } from 'react-navigation';

export interface IProps {
    account: IAccountState;
    blockchain: Blockchain;
    styles: ReturnType<typeof stylesProvider>;
    navigation: NavigationScreenProp<NavigationState, NavigationParams>;
}

export const AccountCardComponent = (props: IProps) => {
    const styles = props.styles;
    return (
        <TouchableOpacity
            testID="account-card"
            style={styles.container}
            onPress={() => {
                props.navigation.navigate('Account');
            }}
        >
            <Icon name="money-wallet-1" size={25} style={styles.icon} />
            <View style={styles.accountInfoContainer}>
                <View style={{ flexDirection: 'row', alignItems: 'baseline' }}>
                    <Text format={{ currency: BLOCKCHAIN_INFO[props.account.blockchain].coin }}>
                        {props.account.balance}
                    </Text>
                    <Text small>
                        {'  '} $
                        <Convert
                            from={BLOCKCHAIN_INFO[props.account.blockchain].coin}
                            to="USD"
                            style={{ fontSize: 12, marginLeft: 82 }}
                            amount={props.account.balance.value}
                        />
                    </Text>
                </View>
                <Text small>{props.account.address} </Text>
            </View>
            <Icon name="arrow-right-1" size={25} style={styles.icon} />
        </TouchableOpacity>
    );
};

export const AccountCard = withTheme(stylesProvider)(AccountCardComponent);
