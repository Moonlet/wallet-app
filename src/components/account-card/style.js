import { StyleSheet } from 'react-native';
import COLORS from '../../styles/colors';

export default StyleSheet.create({
    container: {
        backgroundColor: COLORS.cardBackground,
        borderRadius: 6,
        height: 80,
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginBottom: 8,
        marginHorizontal: 4,
        flexDirection: 'row',
        padding: 12
    },
    accountInfoContainer: {
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
        marginHorizontal: 8
    }
});
