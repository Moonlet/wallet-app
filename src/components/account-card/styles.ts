import { StyleSheet, Platform } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { BASE_DIMENSION, BORDER_RADIUS, ICON_CONTAINER_SIZE } from '../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            backgroundColor: theme.colors.cardBackground,
            borderRadius: BORDER_RADIUS,
            paddingHorizontal: BASE_DIMENSION / 2,
            paddingVertical: BASE_DIMENSION * 2,
            display: 'flex',
            justifyContent: 'space-between',
            alignItems: 'center',
            marginBottom: BASE_DIMENSION,
            flexDirection: 'row'
            // padding: 12
        },
        accountInfoContainer: {
            flex: 1,
            display: 'flex',
            flexDirection: 'column',
            marginHorizontal: BASE_DIMENSION
        },
        iconContainer: {
            width: ICON_CONTAINER_SIZE,
            height: ICON_CONTAINER_SIZE,
            justifyContent: 'center',
            alignItems: 'center'
        },
        icon: {
            color: theme.colors.accent
        },
        amountContainer: {
            flexDirection: 'row',
            alignItems: 'baseline'
        },
        address: {
            fontSize: 16,
            lineHeight: 20,
            color: theme.colors.text,
            opacity: 0.67
        },
        firstAmount: {
            fontSize: 18,
            letterSpacing: 0.38,
            color: theme.colors.text,
            opacity: 0.87,
            fontWeight: '500'
        },
        secondAmount: {
            fontSize: 12,
            letterSpacing: 0.38,
            color: theme.colors.text,
            opacity: 0.87,
            marginLeft: 8,
            fontWeight: '500'
        }
    });
