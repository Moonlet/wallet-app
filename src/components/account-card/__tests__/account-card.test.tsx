import 'react-native';
import React from 'react';
import { AccountCardComponent, IProps } from '../account-card';
import { Blockchain } from '../../../core/blockchain/types';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';

const props: IProps = {
    account: {
        index: 1,
        blockchain: Blockchain.ZILLIQA,
        address: 'zil1vs74hw5k21233h432kj321l3k21b',
        publicKey: '1',
        balance: 12332
    },
    blockchain: Blockchain.ZILLIQA,
    styles: styleProvider(darkTheme)
};
describe('account card component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<AccountCardComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
