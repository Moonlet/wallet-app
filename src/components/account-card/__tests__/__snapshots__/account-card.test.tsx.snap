// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account card component renders correctly 1`] = `
"<TouchableOpacity testID=\\"account-card\\" style={{...}} onPress={[Function: onPress]} activeOpacity={0.2}>
  <View style={{...}}>
    <Icon name=\\"money-wallet-1\\" size={24} style={{...}} allowFontScaling={false} />
  </View>
  <View style={{...}}>
    <View style={{...}}>
      <Connect(AmountComponent) style={{...}} amount={{...}} blockchain=\\"ZILLIQA\\" />
      <Connect(AmountComponent) style={{...}} amount={{...}} blockchain=\\"ZILLIQA\\" convert={true} />
    </View>
    <withTheme(TextComponent) style={{...}}>
      zil1v...3k21b
       
    </withTheme(TextComponent)>
  </View>
  <View style={{...}}>
    <Icon name=\\"arrow-right-1\\" size={18} style={{...}} allowFontScaling={false} />
  </View>
</TouchableOpacity>"
`;
